### PR TITLE
Add Windows CI

### PR DESCRIPTION
**Pull request by certik**
I am trying to troubleshoot #58.

### CODE REVIEW

**Comment by certik** [comment]
Here is the failure:
```
2/3 Testing: load_text
2/3 Test: load_text
Command: "D:/a/stdlib/stdlib/build/src/tests/loadtxt/test_loadtxt.exe" "D:/a/stdlib/stdlib/build/src/tests/loadtxt"
Directory: D:/a/stdlib/stdlib/src/tests/loadtxt
"load_text" start time: Dec 31 23:46 Coordinated Universal Time
Output:
----------------------------------------------------------
 Array, shape=(           4 ,           2 )
   1.00000000       2.00000000    
   3.00000000       4.00000000    
   5.00000000       6.00000000    
   7.00000000       8.00000000    
 Array, shape=(           4 ,           2 )
   1.0000000000000000        2.0000000000000000     
   3.0000000000000000        4.0000000000000000     
   5.0000000000000000        6.0000000000000000     
   7.0000000000000000        8.0000000000000000     
 Array, shape=(           4 ,           3 )
   1.0000000000000000        2.0000000000000000        9.0000000000000000     
   3.0000000000000000        4.0000000000000000        10.000000000000000     
   5.0000000000000000        6.0000000000000000        11.000000000000000     
   7.0000000000000000        8.0000000000000000        12.000000000000000     
 Array, shape=(          16 ,           2 )
   1.0000000000000000E-008   91.999987593924899     
   1.0241132548855634E-008   91.999987314749688     
   1.0482337218958209E-008   91.999987035877282     
   1.0723614031878819E-008   91.999986757297677     
   1.0964963009194818E-008   91.999986479001350     
   1.1206384172490366E-008   91.999986200979166     
   1.1447877543355709E-008   91.999985923222511     
   1.1689443143387538E-008   91.999985645723044     
   1.1931080994189523E-008   91.999985368472906     
   1.2172791117370886E-008   91.999985091464495     
   1.2414573534548370E-008   91.999984814690578     
   1.2656428267344438E-008   91.999984538144247     
   1.2898355337388186E-008   91.999984261818796     
   1.3140354766315149E-008   91.999983985707871     
   1.3382426575767665E-008   91.999983709805363     
   1.3624570787394342E-008   91.999983434105332     
 Array, shape=(           3 ,          10 )
   1.5636717312299885E-010   4.5156817177622978E-007   4.9656862178073029E-006   5.0106866678118064E-005   5.0151867128122533E-004   5.0176362928751987E-003   5.5848764877645951E-002  0.32618374746711520        1.7639051761733842        9.4101331514118236     
   8.2348196112966627E-010   4.5823931965629650E-007   5.0323976966079676E-006   5.0773981466124731E-005   5.0818981916129179E-004   5.0928786314535686E-003   5.6248925898183838E-002  0.32831192218075922        1.7752234390209392        9.4703270222745211     
   2.0220116378489263E-009   4.7022461642348905E-007   5.1522506642798948E-006   5.1972511142843962E-005   5.2017511592848458E-004   5.2280580298917183E-003   5.6967849938248938E-002  0.33213537295325257        1.7955576815764616        9.5784705410250410     

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0xffffffff
#1  0xffffffff
#2  0xffffffff
#3  0xffffffff
#4  0xffffffff
#5  0xffffffff
#6  0xffffffff
#7  0xffffffff
<end of output>
Test time =   0.02 sec
----------------------------------------------------------
Test Failed.
"load_text" end time: Dec 31 23:46 Coordinated Universal Time
"load_text" time elapsed: 00:00:00
----------------------------------------------------------
```

That's weird, it looks like some compiler bug.

**Comment by certik** [comment]
@scivision now all the tests pass! I disabled qp for now.

File: .github/workflows/ci_windows.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    - name: CTest
+      run: ctest --output-on-failure --parallel -V
+      working-directory: build
+
+    - uses: actions/upload-artifact@v1
+      if: failure()
+      with:
+        name: WindowsCMakeTestlog
+        path: build/Testing/Temporary/LastTest.log
```
Comment (zbeekman) [inline .github/workflows/ci_windows.yml]: Small nit: a final new line would be good here. But this is a triviality so ¯\_(ツ)_/¯

Comment (certik) [reply .github/workflows/ci_windows.yml]: Fixed in 6b4d861bced69ecfe806a0d801965e18b8a70576. My vim adds it automatically, but I haven't touched this file.

**Comment by certik** [comment]
Given that the quadruple precision is buggy on Windows, probably the best way forward is to split the tests into a single/double file, and a quadruple file. The quadruple file test would be skipped on Windows.

**Comment by certik** [comment]
@zbeekman do you want to merge this first, so that you can work on consolidating the CI into one file? And I can work on getting quadruple working again at least on Linux and macOS?

**Comment by zbeekman** [comment]
> @zbeekman do you want to merge this first, so that you can work on consolidating the CI into one file?

Sure that sounds good! Mind if I also rebase and merge #64?

**Comment by certik** [comment]
No, that's why we should all allow "edits by maintainers" in our PRs.

Yes, if you can figure out how to consolidate the Windows test into the same matrix, I think this would be the best.

**Comment by certik** [comment]
@zbeekman go ahead and merge this when you are ready.

**Comment by zbeekman** [comment]
> No, that's why we should all allow "edits by maintainers" in our PRs.

I think we're mis-communicating again, sorry, probably my fault. I'll merge this now.

**Comment by certik** [comment]
(My "No" meant "No, I don't mind", as a reply to yours "Mind if I also rebase and merge #64?".)

**Pull request merged**